compiler/rustc_passes/src/stability.rs

```diff
@@ -90,6 +90,26 @@ fn inherit_deprecation(def_kind: DefKind) -> InheritDeprecation {
     }
 }
 
+fn inherit_const_stability(tcx: TyCtxt<'_>, def_id: LocalDefId) -> InheritConstStability {
+    let def_kind = tcx.def_kind(def_id);
+    match def_kind {
+        DefKind::AssocFn | DefKind::AssocTy | DefKind::AssocConst => {
```

Comment:
does this mean that we add consider associated types and consts const unstable if they are in a const stable impl?
should this only be `DefKind::AssocFn`? It seems like this is already the behavior on stable, so I guess it's kinda fine?

Reply:
> does this mean that we add consider associated types and consts const unstable if they are in a const stable impl?
Yes, if it's in a *trait* impl.

> should this only be DefKind::AssocFn? It seems like this is already the behavior on stable, so I guess it's kinda fine?
I don't know enough of const stability. Who monitors const stability in libs?

Reply:
cc @rust-lang/wg-const-eval i don't think const stability on anything apart from impls (for const unstable) and functions makes sense, does it?

Reply:
well, it's strictly safer to only check for associated functions here, but even without that, we never query const stability on anything *but* functions, so removing the other two patterns is correct.